### PR TITLE
Moves goal tracking page calls to the client

### PR DESCRIPTION
**Pull request by discorick**
Adds click handlers to provide "meta-page" calls for GA goal tracking. Closes https://github.com/huboard/huboard-web/issues/313.

### CODE REVIEW

**Comment by discorick** [comment]
image

File: app/views/shared/_analytics.html.erb [hunk 1]
```diff
@@ -2,6 +2,17 @@
 <% if ENV['SEGMENTIO_KEY'] %>
   window.analytics=window.analytics||[],window.analytics.methods=["identify","group","track","page","pageview","alias","ready","on","once","off","trackLink","trackForm","trackClick","trackSubmit"],window.analytics.factory=function(t){return function(){var a=Array.prototype.slice.call(arguments);return a.unshift(t),window.analytics.push(a),window.analytics}};for(var i=0;i<window.analytics.methods.length;i++){var key=window.analytics.methods[i];window.analytics[key]=window.analytics.factory(key)}window.analytics.load=function(t){if(!document.getElementById("analytics-js")){var a=document.createElement("script");a.type="text/javascript",a.id="analytics-js",a.async=!0,a.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.io/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(a,n)}},window.analytics.SNIPPET_VERSION="2.0.9",
   window.analytics.load("<%= ENV['SEGMENTIO_KEY'] %>");
+  var $ = $ ? $ : false;
+  if($){
+    $('a').click(function(e){
```
Comment (dahlbyk) [inline app/views/shared/_analytics.html.erb]: If we don't qualify this selector, I would expect some internal links to be double-tracked.

**Comment by dahlbyk** [comment]
I'm fuzzy how the Authenticated event is firing since there's round-trip to GitHub involved. What am I missing?

**Comment by discorick** [comment]
Oh, we don't really need it anymore. `/dashboard` is the new authenticated. Does that seem reasonable ?

**Comment by dahlbyk** [comment]
> Oh, we don't really need it anymore. `/dashboard` is the new authenticated. Does that seem reasonable ?

:+1:

File: app/views/shared/_analytics.html.erb (hunk 1 above)
Comment (discorick) [inline app/views/shared/_analytics.html.erb]: Watching on testing, I didn't notice any dups, we can qualify though, same as the buttons

**Pull request merged**